src/index.ts

```diff
@@ -162,5 +170,52 @@ export function transposeTarStream(tarStream: NodeJS.ReadableStream,
 
 		tarStream.pipe(extract)
 	})
+}
+
+/**
+ * getBuildThroughStream: Get a through stream, which when piped to will remove all
+ * extra output that is added as a result of this module transposing a Dockerfile.
+ *
+ * This function enables 'silent' emulated builds, with the only difference in output
+ * from a native build being that there is an extra COPY step, where the emulator is
+ * added to the container
+ */
+export function getBuildThroughStream(opts: TransposeOptions): NodeJS.ReadWriteStream {
+
+	// Regex to match against 'Step 1/5:', 'Step 1/5 :' 'Step 1:' 'Step 1 :'
+	// and all lower case versions.
+	const stepLineRegex = /^(?:step)\s\d(?:\/\d)?\s?:/i
+	const isStepLine = RegExp.prototype.test.bind(stepLineRegex)
 
+	// Function to strip the string matched with the regex above
+	const stripStepPrefix = (data: string): string => {
+		return data.substr(data.indexOf(':') + 1)
+	}
+
+	// Regex to match against the type of command, e.g. FROM, RUN, COPY
+	const stepCommandRegex = /^\s?(\w+)(:?\s)/i
+	const getStepCommand = RegExp.prototype.exec.bind(stepCommandRegex)
+
+	// Regex to remove extra flags which this module adds in
+	const replaceRegex = new RegExp(`${opts.containerQemuPath} -execve /bin/sh -c `, 'i')
+	const replaceQemuLine = (data: string): string => {
+		return data.replace(replaceRegex, '')
+	}
+
+	return es.pipe(
+		// Ensure we get one line at a time
+		es.split(),
+		es.through(function(data: string) {
+
+			if (isStepLine(data)) {
+				if (getStepCommand(stripStepPrefix(data))[1].toUpperCase() === 'RUN') {
+					this.emit('data', replaceQemuLine(data) + EOL)
+				} else {
+					this.emit('data', data + EOL)
+				}
+			} else {
+				this.emit('data', data + EOL)
+			}
```

Comment:
Alternatively it can be:
```typescript
if (isStepLine(data) && getStepCommand(stripStepPrefix(data))[1].toUpperCase() === 'RUN') {
	data = replaceQemuLine(data)
}
this.emit('data', data + EOL)
```